jaeger-zipkin/src/main/java/io/jaegertracing/zipkin/V2SpanConverter.java

```diff
@@ -0,0 +1,187 @@
+            .timestamp(span.getStart())
+            .duration(span.getDuration());
+
+    buildAnnotations(span, builder);
+    buildTags(span, builder);
+
+    return builder.build();
+  }
+
+  private static zipkin2.Span.Kind convertKind(Object kind) {
+    if (Tags.SPAN_KIND_SERVER.equals(kind)) {
+      return zipkin2.Span.Kind.SERVER;
+    } else if (Tags.SPAN_KIND_CLIENT.equals(kind)) {
+      return zipkin2.Span.Kind.CLIENT;
+    } else if (Tags.SPAN_KIND_CONSUMER.equals(kind)) {
+      return zipkin2.Span.Kind.CONSUMER;
+    } else if (Tags.SPAN_KIND_PRODUCER.equals(kind)) {
+      return zipkin2.Span.Kind.PRODUCER;
+    } else {
+      return null;
+    }
+  }
+
+  private static void buildAnnotations(Span span, zipkin2.Span.Builder builder) {
+    if (ConverterUtil.isRpc(span)) {
+      String startLabel = zipkin.Constants.SERVER_RECV;
+      String endLabel = zipkin.Constants.SERVER_SEND;
+      if (ConverterUtil.isRpcClient(span)) {
+        startLabel = zipkin.Constants.CLIENT_SEND;
+        endLabel = zipkin.Constants.CLIENT_RECV;
+      }
+
+      builder.addAnnotation(span.getStart(), startLabel);
+      builder.addAnnotation(span.getStart() + span.getDuration(), endLabel);
+    }
+
+    List<LogData> logs = span.getLogs();
+    if (logs != null) {
+      for (LogData logData : logs) {
+        String logMessage = logData.getMessage();
+        Map<String, ?> logFields = logData.getFields();
+        if (logMessage != null) {
+          builder.addAnnotation(logData.getTime(), logMessage);
+        } else if (logFields != null) {
+          builder.addAnnotation(logData.getTime(), gson.toJson(logFields));
+        }
+      }
+    }
+  }
+
+  private static void buildTags(Span span, zipkin2.Span.Builder builder) {
+    Map<String, Object> tags = span.getTags();
+    boolean isRpc = ConverterUtil.isRpc(span);
+    boolean firstSpanInProcess = span.getReferences().isEmpty() || ConverterUtil.isRpcServer(span);
+
+    if (firstSpanInProcess) {
+      Map<String, ?> processTags = span.getTracer().tags();
+      // add tracer tags to first zipkin span in a process but remove "ip" tag as it is
+      // taken care of separately.
+      for (Map.Entry<String, ?> entry : processTags.entrySet()) {
+        String tagKey = entry.getKey();
+        if (!Constants.TRACER_IP_TAG_KEY.equals(tagKey)) {
+          Object tagValue = entry.getValue();
+          // add a tracer. prefix to process tags for zipkin
+          builder.putTag("tracer." + tagKey, tagValue.toString());
+        }
+      }
+    }
+
+    if (!isRpc) {
+      String componentName;
+      Object componentTag = tags.get(Tags.COMPONENT.getKey());
+      if (componentTag instanceof String) {
+        componentName = componentTag.toString();
+      } else {
+        // spans always have associated tracers, and service names
+        componentName = span.getTracer().getServiceName();
+      }
+
+      builder.putTag(zipkin.Constants.LOCAL_COMPONENT, componentName);
```

Comment:
What about this one?  In Zipkin v2 spans, the local endpoint field contains the service name so this seems redundant to add this special tag.

I'm going to remove it for now but let me know if you think it should remain.